src/h5web/visualizations/index.test.ts

```diff
@@ -287,8 +306,253 @@ describe('Visualization definitions', () => {
+        },
+        {
+          ...baseDataset,
+          type: { class: HDF5TypeClass.Float, base: 'H5T_IEEE_F32BE' },
+          shape: { class: HDF5ShapeClass.Simple, dims: [1, 2, 3, 4] },
+          attributes: [
+            { name: 'interpretation', value: 'unknown' } as HDF5Attribute,
+          ], // unknown interpretation
+        },
+      ];
+
+      supportedSignals.forEach((signal) => {
+        const metadata = {
+          datasets: { [baseDataset.id]: signal },
+        } as HDF5Metadata;
+
+        const supported = nxImageDef.supportsEntity(nxDataGroup, metadata);
+        expect(supported).toBe(true);
+      });
+    });
+
+    it('should not support NXdata group referencing signal with non-numeric type, non-simple shape, less than two dimensions or with known, non-image interpretation', () => {
+      const unsupportedSignals: HDF5Entity[] = [
+        {
+          ...baseDataset,
+          type: { class: HDF5TypeClass.Compound, fields: [] }, // type class not supported
+          shape: { class: HDF5ShapeClass.Simple, dims: [4, 15] },
+        },
+        {
+          ...baseDataset,
+          type: { class: HDF5TypeClass.Float, base: 'H5T_IEEE_F64LE' },
+          shape: { class: HDF5ShapeClass.Scalar }, // shape class not supported
+        },
+        {
+          ...baseDataset,
+          type: { class: HDF5TypeClass.Integer, base: 'H5T_STD_I32LE' },
+          shape: { class: HDF5ShapeClass.Simple, dims: [1] }, // dims length not supported
+        },
+        {
+          ...baseDataset,
+          type: { class: HDF5TypeClass.Integer, base: 'H5T_STD_I32LE' },
+          shape: { class: HDF5ShapeClass.Simple, dims: [4, 15] },
+          attributes: [
+            {
+              name: 'interpretation',
+              value: NxInterpretation.Spectrum,
+            } as HDF5Attribute, // known, non-image interpretation
+          ],
+        },
+      ];
+
+      unsupportedSignals.forEach((signal) => {
+        const metadata = {
+          datasets: { [baseDataset.id]: signal },
+        } as HDF5Metadata;
+
+        const supported = nxImageDef.supportsEntity(nxDataGroup, metadata);
+        expect(supported).toBe(false);
+      });
+    });
+
+    it('should not support NXdata group referencing inexistent signal', () => {
+      const metadata = {} as HDF5Metadata;
+      const supported = nxImageDef.supportsEntity(nxDataGroup, metadata);
+      expect(supported).toBe(false);
+    });
+
+    it('should not support NXdata group with missing signal link', () => {
+      const supported = nxImageDef.supportsEntity(
+        { ...nxDataGroup, links: [] },
+        {} as HDF5Metadata
+      );
+
+      expect(supported).toBe(false);
+    });
+
+    it('should not support dataset, datatype and non-NXdata group', () => {
+      const unsupportedEntities: HDF5Entity[] = [
+        { ...baseDataset, type: '', shape: { class: HDF5ShapeClass.Null } },
+        { ...baseDatatype, type: '' },
```

Comment:
What happens if you put a valid `NXdata` attribute to a dataset or a datatype ?

Reply:
I got lazy on this one -- you're right, it's important to check.

Reply:
Hmm, if you don't mind, I'm going to work on mock data generation first (#253), because this is becoming quite unwieldy. Once I have a set of utility functions I can use to generate all kinds of HDF5 models, it will clean up this test file significantly.